heidrun/getty.rb

```diff
@@ -0,0 +1,127 @@
+           .first.value
+  id = p.root[Krikri::PrimoParser.record('control', 'recordid').join('/')]
+       .first.value
+
+  base_url = 'http://primo.getty.edu/primo_library/libweb/action/dlDisplay.do'
+
+  case source
+  when 'GETTY_ROSETTA'
+    "#{base_url}?vid=GRI&afterPDS=true&institution=01GRI&docId=#{id}"
+  when 'GETTY_OCP'
+    "#{base_url}?vid=GRI-OCP&afterPDS=true&institution=01GRI&docId=#{id}"
+  end
+end
+
+lookup_language = lambda do |i|
+  val = i.value.downcase
+  (val == 'multiple languages') ? 'mul' : val
+end
+
+Krikri::Mapper.define(:getty,
+                      parser: Krikri::PrimoParser) do
+  provider class: DPLA::MAP::Agent do
+    uri 'http://dp.la/api/contributor/getty'
+    label 'J. Paul Getty Trust'
+  end
+
+  dataProvider class: DPLA::MAP::Agent,
+               each: record.field(*Krikri::PrimoParser.record('control',
+                                                              'sourceid'))
+                      .map { |i| DATA_PROVIDER_MAP.fetch(i.value, i.value) }
+                      .first_value,
+               as: :provider do
+    label provider
+  end
+
+  isShownAt class: DPLA::MAP::WebResource do
+    uri build_url
+  end
+
+  preview class: DPLA::MAP::WebResource do
+    uri record.field('sear:LINKS', 'sear:thumbnail')
+  end
+
+  sourceResource class: DPLA::MAP::SourceResource do
+    # TODO: Enrichment needed to split title fields on semicolons
+    title record.fields(Krikri::PrimoParser.display('title'),
+                        Krikri::PrimoParser.display('lds03'))
+
+    collection class: DPLA::MAP::Collection,
+               each: record.fields(Krikri::PrimoParser.display('lds34'),
+                                   Krikri::PrimoParser.display('lds43')),
+               as: :coll do
+      title coll
+    end
+
+    # TODO: Enrichment needed to split contributor providedLabel on semicolons
+    contributor class: DPLA::MAP::Agent,
+                each: record.field(*Krikri::PrimoParser.display('contributor')),
+                as: :contrib do
+      providedLabel contrib
+    end
+
+    creator class: DPLA::MAP::Agent,
+            each: record.field(*Krikri::PrimoParser.display('creator')),
+            as: :creator do
+      providedLabel creator
+    end
+
+    # TODO: Enrichment needed here to remove trailing punctuation and set
+    # 'begin' and 'end' dates?
+    date class: DPLA::MAP::TimeSpan,
+         each: record.field(*Krikri::PrimoParser.display('creationdate'))
+           .first_value,
+         as: :created do
+      providedLabel created
+    end
+
+    description record.fields(Krikri::PrimoParser.display('lds04'),
+                              Krikri::PrimoParser.display('lds28'),
+                              Krikri::PrimoParser.display('rights'))
+
+    extent record.field(*Krikri::PrimoParser.display('format'))
+
+    dcformat record.field(*Krikri::PrimoParser.display('lds09'))
+
+    # TODO: Enrichment needed here to map (e.g.) 'Still image' to 'image'
```

Comment:
We should be able to handle this with https://github.com/dpla/KriKri/blob/develop/lib/krikri/enrichments/dcmi_type_map.rb